### PR TITLE
BL-970-ds-i-want-to-see-my-mentees-mentors

### PR DESCRIPTION
**Pull request by JeffDesch**
## Description
This feature branch implements the DS side of the "I want to see my mentees/mentors" User Story.
The API was updated with additional endpoints supporting the creation and deletion of mentor-mentee matches (stored in a new 'Matches' db collection) and retrieval of matching 'Mentee' or 'Mentor' records from a mentor/mentee query. Additionally, specialized query methods were added to the MongoDB class to support the operations of the endpoints.

## Jira Ticket
[BL-970](https://bloomtechlabs.atlassian.net/browse/BL-970)

## Type of Change
- [x] New feature

## Checklist
- [x] My code follows PEP8 style guide
- [x] I have removed unnecessary print statements from my code
- [x] I have made corresponding changes to the documentation if necessary
- [x] My changes generate no errors
- [x] No commented-out code
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

## Loom Video
[Video Here](https://www.loom.com/share/92f29f681a124f14a6793ff247d32b8b)

### CODE REVIEW

**Comment by BrokenShell** [comment]
This is a fantastic contribution to the project. Great work! I can't wait to see it in action.

File: app/routers/match_router.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                                                    {"mentor_id": data["mentor_id"]},
+                                                    {"mentee_ids": data["mentee_id"]})}
+
+
+@Router.post("/delete/match")
+async def delete_match(data: MatchUpdate):
+    """Remove a mentor/mentee pairing from the collection
+    <pre><code>
+    @param data: JSON[MatchUpdate]
+    @return JSON[Boolean] - success or failure of update</pre></code>"""
```
Comment (BrokenShell) [inline app/routers/match_router.py:30]: small type here (i think it should say `@return JSON[Boolean] - success or failure of delete</pre></code>` Be careful of the copy/pasta bug ;)

File: app/routers/match_router.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+Router.db = MongoDB()
+
+
+@Router.post("/create/match")
+async def create_match(data: MatchUpdate):
+    """Insert a new mentor/mentee pairing into the collection
+    <pre><code>
+    @param data: JSON[MatchUpdate]
+    @return JSON[Boolean] - success or failure of update</pre></code>"""
+    data = data.dict()
```
Comment (BrokenShell) [inline app/routers/match_router.py:19]: small type here (i think it should say @return JSON[Boolean] - success or failure of creation</pre></code> Be careful of the copy/pasta bug ;)

File: app/routers/match_router.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+@Router.post("/read/match")
+async def get_match(query: MatchQuery):
+    """Retrieve all matching mentor/mentee objects for a given mentee/mentor
+    <pre><code>
+    @param query: JSON[MatchQuery]
+    @return JSON[Array[Mentor]] | JSON[Array[Mentee]]</pre></code>"""
+    query = query.dict()
+    if query["user_type"] == "mentor":
+        db_col = "Mentees"
```
Comment (BrokenShell) [inline app/routers/match_router.py:45]: is `db_col` actually being used?

File: app/routers/match_router.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+                                                  {"mentee_ids": data["mentee_id"]})}
+
+
+@Router.post("/read/match")
+async def get_match(query: MatchQuery):
+    """Retrieve all matching mentor/mentee objects for a given mentee/mentor
+    <pre><code>
+    @param query: JSON[MatchQuery]
+    @return JSON[Array[Mentor]] | JSON[Array[Mentee]]</pre></code>"""
+    query = query.dict()
```
Comment (BrokenShell) [inline app/routers/match_router.py:43]: you take the query from the input, but then later you overwrite it, not sure what was intended here.

**Review by BrokenShell — CHANGES_REQUESTED** [review]
This is a great start, I left a few notes - nothing major.

**Review by nikolayglushetskiy — APPROVED** [review]
Great job!

**Review by zoebrantley — APPROVED** [review]
Looks great!! Really excited to see this made.

**Pull request merged**